saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistrations.java

```diff
@@ -46,6 +47,7 @@ private RelyingPartyRegistrations() {
 	 * applications will also need to provide a registrationId, like so:
 	 *
 	 * <pre>
+	 *  String metadataLocation = "file:C:\\saml\\metadata.xml"
```

Comment:
I think this JavaDoc should show some different possible values. This is important so that readers don't get the idea that `RelyingPartyRegistrations` is only for file-based resolution.

Something like the following might do:

```
Return a {@link RelyingPartyRegistration.Builder} based off of the given SAML 2.0
Asserting Party (IDP) metadata location.

Valid locations can be classpath- or file-based or they can be HTTP endpoints. Some 
valid endpoints might include:

<pre>
  metadataLocation = "classpath:asserting-party-metadata.xml";
  metadataLocation = "file:asserting-party-metadata.xml";
  metadataLocation = "https://ap.example.org/metadata";
</pre>

Note that by default the registrationId ...
```